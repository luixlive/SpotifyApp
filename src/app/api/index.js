import authenticationApi from './authentication';

export default authenticationApi;
