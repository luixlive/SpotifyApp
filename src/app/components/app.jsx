import PropTypes from 'prop-types';
import React from 'react';

import Header from './header';
import Footer from './footer';
import { SizeDetector } from './hoc';

export const App = props => (
  <div>
    <Header isUserAuthenticated={false} />
    {props.children}
    <Footer />
  </div>
);

App.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default SizeDetector(App);
