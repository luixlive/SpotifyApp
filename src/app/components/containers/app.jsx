import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import { Footer } from './../';
import { Header } from './../containers';
import { loadUser } from './../../actions/user';
import ScreenLoader from './../screen_loader';
import { SizeDetector } from './../hoc';

export class PureApp extends Component {
  componentDidMount() {
    this.props.loadUser();
  }

  renderBody() {
    if (this.props.userLoaded) {
      return this.props.children;
    }
    return <ScreenLoader />;
  }

  render() {
    return (
      <div>
        <Header />
        {this.renderBody()}
        <Footer />
      </div>
    );
  }
}

PureApp.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  loadUser: PropTypes.func.isRequired,
  userLoaded: PropTypes.bool.isRequired,
};

const mapStateToProps = ({ user }) => ({ userLoaded: user.userLoaded });

const mapDispatchToProps = dispatch => ({
  loadUser: () => dispatch(loadUser()),
});

export const ConnectedApp =
  connect(mapStateToProps, mapDispatchToProps)(PureApp);

export default withRouter(SizeDetector(ConnectedApp));
