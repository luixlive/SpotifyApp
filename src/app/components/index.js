import AppConnected from './app';
import AppFooterConnected from './footer';
import AppHeaderConnected from './header';
import LoginConnected from './login';
import Router from './router';
import UserStatsConnected from './user_stats';

export {
  AppConnected,
  AppFooterConnected,
  AppHeaderConnected,
  LoginConnected,
  Router,
  UserStatsConnected,
};
