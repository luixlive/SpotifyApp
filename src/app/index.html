<!DOCTYPE html>
<html>
  <head>
    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-118287841-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-118287841-1');
    </script>

    <!-- Favicon stuff -->
    <link rel="apple-touch-icon" sizes="180x180" href="/favicon/apple-touch-icon.png">
    <link rel="icon" type="image/png" sizes="32x32" href="/favicon/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="16x16" href="/favicon/favicon-16x16.png">
    <link rel="manifest" href="/favicon/site.webmanifest">
    <link rel="mask-icon" href="/favicon/safari-pinned-tab.svg" color="#016936">
    <meta name="msapplication-TileColor" content="#da532c">
    <meta name="theme-color" content="#ffffff">

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Spotify App</title>
  </head>
  <body>
    <div class="app"></div>
  </body>
</html>