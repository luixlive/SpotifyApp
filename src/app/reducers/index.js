import { combineReducers } from 'redux';

import deviceMobile from './device';
import error from './error';
import user from './user';
import userStats from './user_stats';

export const reducers = { deviceMobile, error, user, userStats };

export default combineReducers(reducers);
