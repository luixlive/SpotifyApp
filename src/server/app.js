const bodyParser = require('body-parser');
const config = require('config');
const cookieSession = require('cookie-session');
const express = require('express');
const helmet = require('helmet');
const passport = require('passport');
const path = require('path');
const webpack = require('webpack');
const webpackMiddleware = require('webpack-dev-middleware');
const webpackHotMiddleware = require('webpack-hot-middleware');

const apiRouter = require('./routers');
const logger = require('./../utils/logger');
const swaggerRouter = require('./swagger');
const webpackConfig = require('./../../webpack.config');

const app = express();
const environment = config.util.getEnv('NODE_ENV');

logger.info(`Spotify App started in ${config.util.getEnv('NODE_ENV')} mode`);

require('./services/passport').configurePassport(passport);

app.use(helmet());
app.use(bodyParser.json());
app.use(cookieSession({
  maxAge: 30 * 24 * 60 * 60 * 1000,
  keys: [config.get('COOKIE_KEY')],
}));
app.use(passport.initialize());
app.use(passport.session());

app.use('/api', apiRouter);

/* istanbul ignore next  */
if (environment !== 'production') {
  app.use('/swagger', swaggerRouter);
}

const webpackCompiler = webpack(webpackConfig);
switch (environment) {
  /* istanbul ignore next */
  case 'development':
    app.use(webpackMiddleware(webpackCompiler, {}));
    app.use(webpackHotMiddleware(webpackCompiler));
    break;
  case 'test':
    app.get('/test', (req, res) => {
      res.sendStatus(200);
    });
    break;
  /* istanbul ignore next */
  default:
    app.use(express.static(path.join(__dirname, './../../dist')));
}

app.get('*', (req, res) => {
  res.sendFile(path.resolve(__dirname, './../../dist', 'index.html'));
});

module.exports = app;
