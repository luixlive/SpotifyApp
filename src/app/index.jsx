import { AppContainer } from 'react-hot-loader';
import { applyMiddleware, compose, createStore } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { Provider } from 'react-redux';
import React from 'react';
import ReactDOM from 'react-dom';

import { Router } from './components';
import reducers from './reducers';
import rootSaga from './sagas';
import './style/semantic.less';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducers,
  undefined,
  compose(
    applyMiddleware(sagaMiddleware),
    window.devToolsExtension ? window.devToolsExtension() : f => f,
  ),
);

sagaMiddleware.run(rootSaga);

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <Router />
      </Provider>
    </AppContainer>
    , document.querySelector('.app'),
  );
};

render();

if (module.hot) {
  module.hot.accept('./components/router', () => {
    render();
  });
}
