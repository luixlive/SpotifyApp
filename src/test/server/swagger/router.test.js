import request from 'supertest';

import app from './../../../server/app';
import getRouter from './../../../server/swagger/router';

describe('Server Swagger - Router', () => {
  it('calls swaggerDefault controller', () => {
    const controller = { swaggerDefault: jest.fn(), swaggerJSDoc: jest.fn() };
    const router = getRouter(controller);
    const requestRouter = { url: '/', method: 'GET' };
    router.handle(requestRouter);
    expect(controller.swaggerDefault).toHaveBeenCalledTimes(1);
  });

  it('should return Swagger JSON when GET /swagger/api-docs.json', (done) => {
    request(app)
      .get('/swagger/api-docs.json')
      .then((response) => {
        expect(response.statusCode).toBe(200);
        expect(response.text).toMatchSnapshot();
        done();
      });
  });

  it('should return Swagger UI when GET /swagger/api-docs', (done) => {
    request(app)
      .get('/swagger/api-docs')
      .then((response) => {
        expect(response.text).toMatchSnapshot();
        done();
      });
  });
});
