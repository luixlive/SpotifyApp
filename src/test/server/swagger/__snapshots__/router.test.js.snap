// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Swagger - Router should return Swagger JSON when GET /swagger/api-docs.json 1`] = `"{\\"info\\":{\\"title\\":\\"Spotify Stats\\",\\"version\\":\\"1.0.0\\",\\"description\\":\\"Simple webapp to see your user's Spotify statistics.\\"},\\"swagger\\":\\"2.0\\",\\"paths\\":{\\"/api/authentication/logout\\":{\\"get\\":{\\"tags\\":[\\"Authentication\\"],\\"description\\":\\"Logout from Spotify\\",\\"parameters\\":[{\\"in\\":\\"cookie\\",\\"name\\":\\"session\\",\\"description\\":\\"Session cookie signed\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}},{\\"in\\":\\"cookie\\",\\"name\\":\\"session.sig\\",\\"description\\":\\"Signature key for the Session cookie\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}}],\\"responses\\":{\\"204\\":{\\"description\\":\\"User logged out\\"},\\"401\\":{\\"description\\":\\"No user found\\"}}}},\\"/api/authentication/spotify\\":{\\"get\\":{\\"tags\\":[\\"Authentication\\"],\\"description\\":\\"Get redirected to Spotify's login page, after login event /api/authentication/spotify/callback will be called\\"}},\\"/api/authentication/user\\":{\\"get\\":{\\"tags\\":[\\"Authentication\\"],\\"description\\":\\"Retrieve current user (if there is one)\\",\\"produces\\":[\\"application/json\\"],\\"parameters\\":[{\\"in\\":\\"cookie\\",\\"name\\":\\"session\\",\\"description\\":\\"Session cookie signed\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}},{\\"in\\":\\"cookie\\",\\"name\\":\\"session.sig\\",\\"description\\":\\"Signature key for the Session cookie\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}}],\\"responses\\":{\\"200\\":{\\"description\\":\\"User found and returned\\",\\"schema\\":{\\"$ref\\":\\"#/definitions/User\\"}},\\"401\\":{\\"description\\":\\"No user found\\"}}}},\\"/api/stats/topArtists\\":{\\"get\\":{\\"tags\\":[\\"Stats\\"],\\"description\\":\\"Retrieve from Spotify user's top artists\\",\\"parameters\\":[{\\"in\\":\\"cookie\\",\\"name\\":\\"session\\",\\"description\\":\\"Session cookie signed\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}},{\\"in\\":\\"cookie\\",\\"name\\":\\"session.sig\\",\\"description\\":\\"Signature key for the Session cookie\\",\\"default\\":\\"Set it in the browser, leave this field as it is\\",\\"schema\\":{\\"type\\":\\"string\\"}},{\\"in\\":\\"query\\",\\"name\\":\\"limit\\",\\"description\\":\\"Number of artists (minimum 1, maximum 50)\\",\\"default\\":5,\\"type\\":\\"number\\",\\"format\\":\\"int32\\",\\"minimum\\":1,\\"maximum\\":50},{\\"in\\":\\"query\\",\\"name\\":\\"offset\\",\\"description\\":\\"Offset value (minimum 0)\\",\\"default\\":0,\\"type\\":\\"number\\",\\"format\\":\\"int32\\",\\"minimum\\":0},{\\"in\\":\\"query\\",\\"name\\":\\"timeRange\\",\\"description\\":\\"Time range to take into account artists\\",\\"type\\":\\"string\\",\\"default\\":\\"medium_term\\",\\"enum\\":[\\"long_term\\",\\"medium_term\\",\\"short_term\\"]}],\\"responses\\":{\\"200\\":{\\"description\\":\\"List of user's top artists\\",\\"schema\\":{\\"$ref\\":\\"#/definitions/Artists\\"}},\\"400\\":{\\"description\\":\\"Bad request\\",\\"schema\\":{\\"$ref\\":\\"#/definitions/Error\\"}},\\"401\\":{\\"description\\":\\"No user found\\"},\\"502\\":{\\"description\\":\\"Error retrieving artists from Spotify\\",\\"schema\\":{\\"$ref\\":\\"#/definitions/Error\\"}}}}}},\\"definitions\\":{\\"Error\\":{\\"type\\":\\"object\\",\\"required\\":[\\"error\\"],\\"properties\\":{\\"error\\":{\\"type\\":\\"string\\"}}},\\"UserBase\\":{\\"type\\":\\"object\\",\\"required\\":[\\"externalUrls\\",\\"followers\\",\\"href\\",\\"id\\",\\"images\\",\\"type\\",\\"uri\\"],\\"properties\\":{\\"externalUrls\\":{\\"type\\":\\"object\\",\\"properties\\":{\\"spotify\\":{\\"type\\":\\"string\\",\\"format\\":\\"uri\\"}}},\\"followers\\":{\\"type\\":\\"object\\",\\"properties\\":{\\"href\\":{\\"type\\":\\"string\\",\\"format\\":\\"uri\\"},\\"total\\":{\\"type\\":\\"number\\",\\"format\\":\\"int64\\"}}},\\"href\\":{\\"type\\":\\"string\\",\\"format\\":\\"uri\\"},\\"id\\":{\\"type\\":\\"string\\"},\\"images\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"object\\",\\"properties\\":{\\"height\\":{\\"type\\":\\"number\\",\\"format\\":\\"int32\\"},\\"url\\":{\\"type\\":\\"string\\",\\"format\\":\\"uri\\"},\\"width\\":{\\"type\\":\\"number\\",\\"format\\":\\"int32\\"}}}},\\"type\\":{\\"type\\":\\"string\\"},\\"uri\\":{\\"type\\":\\"string\\",\\"format\\":\\"uri\\"}}},\\"User\\":{\\"type\\":\\"object\\",\\"required\\":[\\"accessToken\\",\\"profile\\",\\"refreshToken\\"],\\"properties\\":{\\"accessToken\\":{\\"type\\":\\"string\\"},\\"profile\\":{\\"allOf\\":[{\\"$ref\\":\\"#/definitions/UserBase\\"},{\\"type\\":\\"object\\",\\"properties\\":{\\"displayName\\":{\\"type\\":\\"string\\"}}}]},\\"refreshToken\\":{\\"type\\":\\"string\\"}}},\\"Artists\\":{\\"type\\":\\"array\\",\\"items\\":{\\"required\\":[\\"genres\\",\\"name\\",\\"popularity\\"],\\"allOf\\":[{\\"$ref\\":\\"#/definitions/UserBase\\"},{\\"type\\":\\"object\\",\\"properties\\":{\\"genres\\":{\\"type\\":\\"array\\",\\"items\\":{\\"type\\":\\"string\\"}},\\"name\\":{\\"type\\":\\"string\\"},\\"popularity\\":{\\"type\\":\\"number\\",\\"format\\":\\"int32\\"}}}]}}},\\"responses\\":{},\\"parameters\\":{},\\"securityDefinitions\\":{},\\"tags\\":[{\\"name\\":\\"Authentication\\"},{\\"name\\":\\"Stats\\"}]}"`;

exports[`Server Swagger - Router should return Swagger UI when GET /swagger/api-docs 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
<head>
<meta charset=\\"utf-8\\">
<title>Redirecting</title>
</head>
<body>
<pre>Redirecting to <a href=\\"/swagger/api-docs/\\">/swagger/api-docs/</a></pre>
</body>
</html>
"
`;
